### ISSUE TITLE
Generate data dictionary include_prerelease fails when a field has an external dependency

### ISSUE THREAD

**Issue by georgeslater** [SFDO-Tooling/CumulusCI#2693]
**Describe the bug**
The new `include_prerelease` option for `generate_data_dictionary` returns an error when a CustomField has a `referenceTo` an object that is not in the package, such as (for an NPSP dependent package) `npe03__Recurring_Donation__c`.  The error is: `referenceTo not found in CustomField`

**Reproduction steps**
1. Have a custom field in your package with `referenceTo` `npe03__Recurring_Donation__c` 
2. Run `cci task run generate_data_dictionary --include_prerelease True`

**Context**
Please include the following information at minimum.

- Your CumulusCI version (`cci version`). **3.38**
- Your Python version (`python --version`) **2.7.16**
- Your operating system and version. **macOS Big Sur 11.2.1**
- If possible, whether you installed CumulusCI using `brew`, `pip`, or `pipx`. **brew**

**Gist**
If applicable, please use the `cci error gist` command to create a Gist from your error log and include the link here. See the [documentation](https://cumulusci.readthedocs.io/en/latest/features.html#working-with-errors) to learn how to create a Gist.   **https://gist.github.com/georgeslater/d6b8ab5d75878c767af7a3347d74eb7b**

**Additional information**
Ideally I would be able to pass in `--include_dependencies True` to fix this but this is explicitly prohibited by the command

**Comment by georgeslater** [comment]
@davidmreed This is an example of a lookup field with an external referenceTo - https://gist.github.com/georgeslater/eaa9d9bff6f2803389fd22ddda2c966b.   Our project is in SFDX format.

I have no conclusive proof it is the culprit but it would seem a likely candidate.   We have a number of fields with NPSP dependencies like this one.  We haven't yet released a full version so it is difficult to debug!

**Comment by davidmreed** [comment]
George, I'm having a hard time reproducing this issue. I tested the data dictionary generator on a project that has lookups to fields owned by its upstream and it worked just fine. I can't see in the code any way to produce this error without having metadata that's actually not valid - i.e., is declared as a lookup but _does not in fact have a `referenceTo` property_.

Are you comfortable altering your local CumulusCI to surface more debugging information? At `datadictionary.py` line 517 (just after `elif field_type == "Lookup":`), you could add

    self.logger.warning(f"Attempting to process lookup {field_name} on {sobject}")

which should get you a pointer to which actual metadata file is the proximate cause of the failure.

**Comment by georgeslater** [comment]
@davidmreed Thank you for investigating - I added the debugging as requested, and now I can see that the culprit is the standard ContactId field on the Case object:

```
<?xml version="1.0" encoding="UTF-8"?>
<CustomField xmlns="http://soap.sforce.com/2006/04/metadata">
    <fullName>ContactId</fullName>
    <trackFeedHistory>false</trackFeedHistory>
    <trackHistory>true</trackHistory>
    <trackTrending>false</trackTrending>
    <type>Lookup</type>
</CustomField>
```

So apologies for leading you down the wrong path earlier.

Interestingly, it is not that the metadata was incorrectly committed - when that field is retrieved from a scratch org it is as above, with no `referenceTo`.   Perhaps the command's error output can be improved for cases like this, but please feel free to close this particular bug.

**Comment by davidmreed** [comment]
Will do. Thanks for tracking it down, @georgeslater. There may be a filter missing from the task that would block such fields, which are usually not retrieved.

**Comment by davidmreed** [comment]
Oh, yeah, there's a genuine bug here, it's just not related to the external dependency. The conditional at `datadictionary.py` line 473 is always `true` for products that have a namespace. Will fix the underlying bug as W-9531065 and close out this issue.
